server/game/effects.js

```diff
@@ -305,6 +305,18 @@ const Effects = {
             }
         };
     },
+    addIcons: function(iconsFunc) {
+        return {
+            apply: function(card) {
+                let icons = iconsFunc();
+                _.each(icons, icon => card.addIcon(icon));
+            },
+            unapply: function(card) {
+                let icons = iconsFunc();
```

Comment:
For both this and `addKeywords`, you need to store off which icons / keywords were applied, and use that list to remove them. And like other dynamic effects, it needs to be marked as state dependent. Otherwise, the effect won't be recalculated as new Fool cards come into play, and won't be properly removed as Fool cards leave play.